src/main/kotlin/com/ampnet/auditornode/script/api/classes/WebSocketInput.kt

```diff
@@ -0,0 +1,94 @@
+package com.ampnet.auditornode.script.api.classes
+
+import com.ampnet.auditornode.model.websocket.InputField
+import com.ampnet.auditornode.model.websocket.InputType
+import com.ampnet.auditornode.model.websocket.ReadBooleanCommand
+import com.ampnet.auditornode.model.websocket.ReadFieldsCommand
+import com.ampnet.auditornode.model.websocket.ReadNumberCommand
+import com.ampnet.auditornode.model.websocket.ReadStringCommand
+import com.ampnet.auditornode.model.websocket.WebSocketApi
+import com.ampnet.auditornode.script.api.model.MapApi
+import org.graalvm.polyglot.HostAccess.Export
+import org.graalvm.polyglot.Value
+import java.util.concurrent.LinkedBlockingQueue
+
+class WebSocketInput(private val webSocketApi: WebSocketApi) : Input {
+
+    private val queue = LinkedBlockingQueue<String>()
+
+    private fun Value.getStringMember(name: String): String? {
+        if (!hasMember(name)) {
+            return null
+        }
+
+        val member = getMember(name)
+
+        return if (member.isString) {
+            member.asString()
+        } else {
+            null
+        }
+    }
+
+    private fun Value.asInputField(): InputField? {
+        val type = getStringMember("type")?.let { InputType.parse(it) }
+        val name = getStringMember("name")
+        val description = getStringMember("description")
```

Comment:
Can we extract this values: "type", "name", "description"? It can be useful if they are referenced somewhere else.

Reply:
If that does become the case in the future, I will extract them. I do not see that becoming the case, however, since this logic is specific to the `WebSocketInput` class.